docs/design/coreclr/jit/lsra-detail.md

```diff
@@ -630,7 +688,30 @@ LinearScanAllocation(List<RefPosition> refPositions)
 
 -   Resolution
 
-    -   Perform resolution at block boundaries, adding moves as needed.
+    Since register allocation is performed over a linearized list of nodes, there
+    is no guarantee that register assignments on off-path edges will be consistent.
+    The resolution phase is responsible for performing moves, as needed to match up
+    the register assignments across edges.
+
+    -    The critical edges are handled first. These are the most problematic, as there
+         is no single location at which the moves can be added. We handle the critical
```

Comment:
This sentence doesn't sound right. Did you mean "We handle all the critical edges that originates from the source block"?

Reply:
No, all the critical edges are handled first. I'll work on a rephrasing of this.